### PR TITLE
General: Bump versions everywhere

### PR DESCRIPTION
**Pull request by kraftbj**
Sets up the 7.3 development cycle, updates node required version, and WP minimum version.
#### Changes proposed in this Pull Request:
* 7.2-alpha -> 7.3-alpha, node 8 -> 10, WP minimum 4.9->5.0 in the one place we missed it.

#### Testing instructions:
* n/a

#### Proposed changelog entry for your changes:
* n/a

### CODE REVIEW

**Review by brbrr — DISMISSED** [review]
Good to go! 🚢

File: package.json [hunk 1]
```diff
@@ -203,7 +203,7 @@
     "webpack-cli": "3.2.3"
   },
   "engines": {
-    "node": ">=8",
+    "node": ">=10",
```
Comment (simison) [inline package.json]: ❤️

Comment (simison) [reply package.json]: There should be a way to get Renovate take care of this and [`.nvmrc`](https://github.com/Automattic/jetpack/blob/21f90723048ef8a074da91d33650c4a94644d266/.nvmrc) btw. cc @brbrr

Comment (brbrr) [reply package.json]: It seems like node updates are enabled by default in renovate. It uses `lts` policy by default 8 and 10 are considered an `lts` versions. We can update the policy to use the latest `lts` version.

Comment (brbrr) [reply package.json]: https://github.com/Automattic/jetpack/pull/11716

**Comment by kraftbj** [comment]
Going to wait to merge to see what Renovate does when we push the update policy.

**Review by oskosk — DISMISSED** [review]
LGTM!

**Review by jeherve — DISMISSED** [review]

**Comment by jeherve** [comment]
Needs a rebase now, though.

**Review by oskosk — APPROVED** [review]
LGTM!

**Pull request merged**